android/src/main/java/com/amplitude/android/Amplitude.kt

```diff
@@ -112,6 +112,17 @@ open class Amplitude(
         }
     }
 
+    internal fun setDeviceId(deviceId: String, expectedCurrentDeviceId: String?): Amplitude {
+        amplitudeScope.launch(amplitudeDispatcher) {
+            isBuilt.await()
+            val currentDeviceId = idContainer.identityManager.getIdentity().deviceId
+            if (currentDeviceId == expectedCurrentDeviceId) {
```

Comment:
Is the point of this check so that if the `deviceId` has changed (externally) before this method is run, then skip updating it?

I feel like that could have some edge cases. e.g. the deviceId is 1) set to another value then 2) reset to the "expected" value.

Can you think of a way to do this purely by syncronizing/queueing the calls without being reliant on the "expected value"?

Reply:
Re-written.